pkg/datastore/applicationstore.go

```diff
@@ -54,6 +55,52 @@ func (a *applicationCollection) GetUpdatableShard() (Shard, error) {
 	}
 }
 
```

Comment:
Please add a TODO to generate those kinds of code automatically by specifying some field in the proto file.

https://github.com/pipe-cd/pipecd/blob/master/pkg/model/application.proto#L24

For example,

``` proto
message Application {
    // The generated unique identifier.
    string id = 1 [(validate.rules).string.min_len = 1, shard=client];

```

Reply:
Nice idea 👍